nixos/modules/system/activation/switch-to-configuration.pl

```diff
@@ -65,7 +69,7 @@
 sub getActiveUnits {
     # FIXME: use D-Bus or whatever to query this, since parsing the
     # output of list-units is likely to break.
-    my $lines = `LANG= systemctl list-units --full --no-legend`;
+    my $lines = `LANG= @systemd@/bin/systemctl list-units --full --no-legend`;
```

Comment:
It worries me a bit what will happen if the systemd version used here doesn't match the running systemd.

Perhaps use `/run/current-system/sw/bin/systemctl` here.

Reply:
> Perhaps use /run/current-system/sw/bin/systemctl here.

If doing that, please add a comment above that command line saying why.

Reply:
There are many spots in this file which use `@systemd@/bin/systemctl` already. Do you think I should switch all of them?

Reply:
AFAICT all of them (except one `systemd-escape` call, which presumably doesn't need to contact the daemon anyway) are run for the purposes of talking to the upgraded systemd, so those ones should be correct.

Although, at that point the activation script has been run and changed `/run/current-system` to point to the new system (incl. the new systemd), so theoretically it would still work. But sounds somewhat magical to me.

Reply:
In such a case, probably the best solution is to use result of `sudo readlink /proc/1/exe` to get absolute path to previous systemctl and use it before the `daemon-reexec` command, and after the call use `@systemd@/bin/systemctl` path. I'll make it so.

Reply:
Nah, this is too complex for very little benefit. I will just use `/run/current-system/sw/bin/systemctl` before the `daemon-rexec` call

Reply:
Done.